### PR TITLE
[Access] Get account bugfix with tests

### PR DESCRIPTION
**Pull request by sideninja**
Closes: #4824 

This PR fixes getting an account by making the indexer handle a not found register in the storage and returning nil value instead of an error, which was the culprit for the bug.

Tests were also added to detect such a bug and existing script test suite was improved.

### CODE REVIEW

File: integration/tests/access/access_api_test.go [hunk 1]
```diff
@@ -193,6 +194,14 @@ func (s *AccessAPISuite) testGetAccount(client *client.Client) {
 		s.Assert().Equal(serviceAddress, account.Address)
 		s.Assert().NotZero(serviceAddress, account.Balance)
 	})
+
+	s.Run("get newly created account", func() {
+		addr, err := utils.CreateFlowAccount(s.ctx, s.serviceClient)
+		s.Require().NoError(err)
+		acc, err := client.GetAccount(s.ctx, addr)
+		s.Require().NoError(err)
+		s.Assert().Equal(addr, acc.Address)
```
Comment (peterargue) [inline integration/tests/access/access_api_test.go:203]: maybe also check that the balance is greater than 0

File: module/execution/scripts.go [hunk 2]
```diff
@@ -61,7 +61,7 @@ type Scripts struct {
 
 func NewScripts(
 	log zerolog.Logger,
-	metrics *metrics.ExecutionCollector,
+	metrics module.ExecutionMetrics,
```
Comment (janezpodhostnik) [inline module/execution/scripts.go:64]: Is this related?

File: module/execution/scripts_test.go [hunk 3]
```diff
@@ -3,148 +3,209 @@ package execution
...
-	account, err := scripts.GetAccountAtBlockHeight(context.Background(), address, first.Header.Height)
-	require.NoError(t, err)
-	assert.Equal(t, address, account.Address)
-	assert.NotZero(t, account.Balance)
-	assert.NotZero(t, len(account.Contracts))
+	s.Run("Get New Account", func() {
+		address := s.createAccount()
+		account, err := s.scripts.GetAccountAtBlockHeight(context.Background(), address, s.height)
+		s.Require().NoError(err)
+		s.Require().Equal(address, account.Address)
```
Comment (peterargue) [inline module/execution/scripts_test.go:88]: same here

**Review by janezpodhostnik — APPROVED** [review]

**Review by peterargue — APPROVED** [review]

File: module/execution/scripts.go (hunk 2 above)
Comment (sideninja) [reply module/execution/scripts.go:64]: it was for purpose of testing, but context wise no, so if you think it's better I can make a separate PR.

File: module/execution/scripts_test.go (hunk 3 above)
Comment (sideninja) [reply module/execution/scripts_test.go:88]: 717f24c1abb714f981a0149ea450fe54d6a9511b

File: integration/tests/access/access_api_test.go (hunk 1 above)
Comment (sideninja) [reply integration/tests/access/access_api_test.go:203]: 717f24c1abb714f981a0149ea450fe54d6a9511b

**Review by koko1123 — APPROVED** [review]
Nice work!

**Comment by peterargue** [comment]
@sideninja some of those failures looked legit. e.g. `TestAccessAPI/TestScriptExecutionAndGetAccounts/get_newly_created_account` failed

**Comment by sideninja** [comment]
Missed those, I thought it was a flakey test.

Those tests were failing because using a nil value in an empty slice becomes this mess with `[][]byte{[]byte(nil)}` and then the adapter would have to check if the value is actually nil and I just hated doing that as it felt so fragile, so I decided to change the `RegisterValues` accepting multiple IDs and returning multiple values to only accept one and return one or nil in that case. I know this could be another PR but it actually tightly relates to handling nil values in return which without this would be more fragile. cc @zhangchiqing @peterargue

**Review by zhangchiqing — APPROVED** [review]

File: integration/tests/access/access_api_test.go [hunk 4]
```diff
@@ -153,13 +154,13 @@ func (s *AccessAPISuite) TestScriptExecutionAndGetAccounts() {
 
 	// Run tests against Access 2, which uses local storage
 	s.testGetAccount(s.an2Client)
-	s.testExecuteScriptWithSimpleScript(s.an2Client)
-	s.testExecuteScriptWithSimpleContract(s.an2Client, targetHeight)
+	//s.testExecuteScriptWithSimpleScript(s.an2Client)
```
Comment (peterargue) [inline integration/tests/access/access_api_test.go:157]: is this for testing?

**Review by peterargue — COMMENTED** [review]
tests are still failing, though it's hard to dig out the reason. here's the message:
```
    access_api_test.go:200: 
        	Error Trace:	/home/runner/work/flow-go/flow-go/integration/tests/access/access_api_test.go:200
        	            				/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.8.4/suite/suite.go:112
        	Error:      	Received unexpected error:
        	            	failed to create account: failed to create new account [Error Code: 1007] error caused by: 1 error occurred:
        	            		* checking sequence number failed: [Error Code: 1007] invalid proposal key: public key 0 on account f8d6e0586b0a20c7 has sequence number 2, but given 0
        	            	
        	Test:       	TestAccessAPI/TestScriptExecutionAndGetAccounts/get_newly_created_account
```

File: integration/tests/access/access_api_test.go (hunk 4 above)
Comment (sideninja) [reply integration/tests/access/access_api_test.go:157]: Sorry forgot to remove it.
